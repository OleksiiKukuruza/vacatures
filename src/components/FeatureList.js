import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

export const StyledFeatures = styled.div`
  display: flex;
  flex-direction: column;
  margin: ${props => 2 * props.theme.spacingUnit}px 0;
`;

const FeatureList = ({ features }) => (
  <StyledFeatures>
    {features.map(feature => (
      <div key={feature}>
        > {feature}
      </div>
    ))}
  </StyledFeatures>
);

FeatureList.propTypes = {
  features: PropTypes.arrayOf(PropTypes.string).isRequired
};

export default FeatureList;
