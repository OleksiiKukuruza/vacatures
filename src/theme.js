const theme = {
  primary: {
    main: '#54b6c8',
    contrastText: '#fff'
  },
  background: '#f1f4f5',
  spacingUnit: 8
};

export default theme;
