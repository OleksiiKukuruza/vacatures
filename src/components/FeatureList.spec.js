import React from 'react';
import { shallow } from 'enzyme';
import theme from '../theme';
import FeatureList, { StyledFeatures } from './FeatureList';

const setup = (Component, props) => {
  const wrapper = shallow(<Component {...props} />);

  return {
    wrapper
  };
};

describe('FeatureList', () => {
  it('renders FeatureList without crashing', () => {
    const props = {
      features: ['First', 'Second']
    };
    const { wrapper } = setup(FeatureList, props);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders StyledFeatures without crashing', () => {
    const { wrapper } = setup(StyledFeatures, { theme });
    expect(wrapper).toMatchSnapshot();
  });
});
