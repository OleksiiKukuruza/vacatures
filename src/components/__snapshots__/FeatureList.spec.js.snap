// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FeatureList renders FeatureList without crashing 1`] = `
<styled.div>
  <div
    key="First"
  >
    &gt; 
    First
  </div>
  <div
    key="Second"
  >
    &gt; 
    Second
  </div>
</styled.div>
`;

exports[`FeatureList renders StyledFeatures without crashing 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin: 16px 0;
}

<div
  className="c0"
/>
`;
